components/inputStates.js

```diff
@@ -0,0 +1,24 @@
+// inportStates.js
+
+var _ = require('underscore');
+var Lm = require('./Lm');
+
+var inportStates = [];
+
+// Returns the vni associated with the specified vnid.
+module.exports = function( inportName, sourceNodeName, sourcePortName, newState ) { 
+ 
+   if ( arguments.length === 1 || arguments.length === 3 ) { 
+       if ( _.isUndefined( inportStates[inportName] ) ) { 
+           return;
+       } 
```

Comment:
Unless I'm missing something, I think lines 12-14 can be removed, since undefined would be returned anyway by line 15 if inportStates[inportName] is undefined.